resources/assets/js/components/FindSkillsModal.tsx

```diff
@@ -0,0 +1,481 @@
+                                  )
+                                : [...expandedAccordions, key],
+                            )
+                          }
+                        >
+                          <p data-h2-font-weight="b(700)">{name[locale]}</p>
+                        </Accordion.Btn>
+                        <p
+                          data-h2-padding="b(top, .25) b(bottom, 1) b(right, .5)"
+                          data-h2-font-color="b(black)"
+                          data-h2-font-size="b(small)"
+                          style={{ paddingLeft: "5rem" }}
+                        >
+                          {description[locale]}
+                        </p>
+                        <Accordion.Content>
+                          <ul
+                            data-h2-padding="b(all, 0)"
+                            className="no-list-style-type"
+                          >
+                            {childrenSkillCategories.map(
+                              (childSkillCatergory) => {
+                                return (
+                                  <li key={childSkillCatergory.key}>
+                                    <div
+                                      data-h2-dialog-actions
+                                      data-h2-grid="b(middle, expanded, flush, 0)"
+                                      data-h2-margin="b(right, .5)"
+                                    >
+                                      <div
+                                        data-h2-align="b(left)"
+                                        data-h2-grid-item="b(5of6)"
+                                      >
+                                        <button
+                                          data-h2-button=""
+                                          type="button"
+                                          onClick={() => {
+                                            setFirstVisit(false);
+                                            setButtonClicked(
+                                              childSkillCatergory.key,
+                                            );
+                                            setResultsSectionText({
+                                              name:
+                                                childSkillCatergory.name[
+                                                  locale
+                                                ],
+                                              description:
+                                                childSkillCatergory.description[
+                                                  locale
+                                                ],
+                                            });
+                                            setSkillsResults(
+                                              childSkillCatergory.skills,
+                                            );
```

Comment:
Right now we're setting all these state hooks at the same time, and then clearing them all at the same time when the back button is clicked. It makes me think there might be some redundancy here. Like it might be simpler, logic-wise, to just use one state hook and then compute the other values, something like:
```
setSelectedCategory(childSkillCategory)
resultsSectionText.name = {name: selectedCategory.name[locale], ...}
skillResults = selectedCategory.skills
firstVisit = selectedCategory === null
```

Reply:
That's an excellent idea. so would the other values just be props or variables within the component?

Reply:
The other values would be constants defined somewhere below where the SelectedCategory state hook is defined, and above the return statement.